File: android/SwiftKotlination/features/story/src/test/kotlin/fr/jhandguy/story/factory/mocks/StoryFactoryMock.kt
package fr.jhandguy.story.factory.mocks

import android.app.Activity
import fr.jhandguy.image.model.ImageManagerInterface
import fr.jhandguy.story.coordinator.StoryCoordinatorInterface
import fr.jhandguy.story.factory.StoryFactory
import fr.jhandguy.story.model.Story
import fr.jhandguy.story.model.StoryManagerInterface
import fr.jhandguy.story.viewmodel.StoryViewModel

class StoryFactoryMock(
    val storyManager: StoryManagerInterface,
    val imageManager: ImageManagerInterface,
    val storyCoordinator: StoryCoordinatorInterface
) : StoryFactory {
    override fun makeStoryManager(story: Story): StoryManagerInterface = storyManager
    override fun makeStoryViewModel(story: Story): StoryViewModel = StoryViewModel(this, story)
    override fun makeStoryCoordinator(activity: Activity): StoryCoordinatorInterface = storyCoordinator
    override fun makeImageManager(): ImageManagerInterface = imageManager
}

File: ios/Modules/TestKit/TestKit/Sources/Mocks/Features/Story/Factory/StoryFactoryMock.swift
import ImageKit
import StoryKit

public struct StoryFactoryMock {
    public let storyManager: StoryManagerMock
    public let imageFactory: ImageFactoryMock

    public init(storyManager: StoryManagerMock = StoryManagerMock(), imageManager: ImageManagerMock = ImageManagerMock()) {
        self.storyManager = storyManager
        imageFactory = ImageFactoryMock(imageManager: imageManager)
    }
}

extension StoryFactoryMock: StoryFactory {
    public func makeStoryManager(for _: Story) -> StoryManagerProtocol {
        storyManager
    }

    public func makeStoryViewController(for story: Story) -> StoryViewController {
        let viewController = StoryViewController()
        viewController.viewModel = StoryViewModel(factory: self, story: story)

        return viewController
    }
}

extension StoryFactoryMock: ImageFactory {
    public func makeImageManager() -> ImageManagerProtocol {
        imageFactory.makeImageManager()
    }
}
